django/contrib/admin/sites.py

```diff
@@ -90,7 +90,15 @@ def register(self, model_or_iterable, admin_class=None, **options):
                 )
 
             if model in self._registry:
-                raise AlreadyRegistered('The model %s is already registered' % model.__name__)
+                registered_admin = self._registry[model]
+                registered_model = registered_admin.model
+                raise AlreadyRegistered(
+                    'Cannot register %s for %s, because %s is already registered.' % (
+                        model._meta.label,
+                        registered_admin,
```

Comment:
In the case of `admin.site.register(Model)` (without a `ModelAdmin` subclass, this shows something like `polls.ModelAdmin`. which could be confusing.